src/main/java/de/muenchen/isi/api/validation/HasFoerdermixRequiredSumValidator.java

```diff
@@ -27,14 +29,14 @@ public boolean isValid(final FoerdermixDto value, final ConstraintValidatorConte
         if (value == null) {
             return true;
         }
-        final BigDecimal sumFoerdermix = ObjectUtils
-            .defaultIfNull(value.getAnteilFreifinanzierterGeschosswohnungsbau(), BigDecimal.ZERO)
-            .add(ObjectUtils.defaultIfNull(value.getAnteilGefoerderterMietwohnungsbau(), BigDecimal.ZERO))
-            .add(ObjectUtils.defaultIfNull(value.getAnteilMuenchenModell(), BigDecimal.ZERO))
-            .add(ObjectUtils.defaultIfNull(value.getAnteilPreisgedaempfterMietwohnungsbau(), BigDecimal.ZERO))
-            .add(ObjectUtils.defaultIfNull(value.getAnteilKonzeptionellerMietwohnungsbau(), BigDecimal.ZERO))
-            .add(ObjectUtils.defaultIfNull(value.getAnteilBaugemeinschaften(), BigDecimal.ZERO))
-            .add(ObjectUtils.defaultIfNull(value.getAnteilEinUndZweifamilienhaeuser(), BigDecimal.ZERO));
+        BigDecimal sumFoerdermix = BigDecimal.ZERO;
+
+        List<FoerderartDto> foerderarten = value.getFoerderarten();
+        for (FoerderartDto foederart : foerderarten) {
+            var tmp = ObjectUtils.defaultIfNull(foederart.getAnteilProzent(), BigDecimal.ZERO);
```

Comment:
Die Variable `tmp` besser in `summand` umbenennen und final setzen. Da diese zur `sumFoerdermix` addiert wird.

Alternativ kann auch ein Stream verwendet werden.

```java
        final BigDecimal sumFoerdermix = foerderarten.stream()
                .map(foerderart -> ObjectUtils.defaultIfNull(foerderart.getAnteilProzent(), BigDecimal.ZERO))
                .reduce(BigDecimal.ZERO, BigDecimal::add);

```